source/memory/memory.go

```diff
@@ -195,12 +195,19 @@ func readNdDeviceInfo(path string) nfdv1alpha1.InstanceFeature {
 	return *nfdv1alpha1.NewInstanceFeature(attrs)
 }
 
-func getNumberOfLinesFromFile(path string) (int, error) {
+func getNumberOfNonEmptyLinesFromFile(path string) (int, error) {
 	data, err := os.ReadFile(path)
 	if err != nil {
 		return 0, err
 	}
-	return len(strings.Split(string(data), "\n")), nil
+	content := strings.Split(string(data), "\n")
+	length := 0
+	for _, line := range content {
+		if line != "" {
```

Comment:
Maybe a bit over-cautious, but how about trimming any whitespace from the lines, ignoring "empty" lines that only have whitespace, WDYT?

```suggestion
	length := 0
	for _, line := range strings.Split(string(data), "\n") {
		if strings.TrimSpace(line) != "" {
```

Reply:
yeah, that's a valid point even though I don't expect it to be the case. Gonna update that